desktop_server/src/com/mygdx/game/server/network/EndTurnService.java

```diff
@@ -0,0 +1,82 @@
+package com.mygdx.game.server.network;
+
+import com.artemis.World;
+import com.badlogic.gdx.utils.Json;
+import com.mygdx.game.core.network.messages.ChangeTurnMessage;
+import com.mygdx.game.server.ecs.entityfactory.ComponentFactory;
+import com.mygdx.game.server.model.Client;
+import com.mygdx.game.server.model.GameRoom;
+import io.vertx.core.buffer.Buffer;
+import lombok.NonNull;
+import lombok.extern.java.Log;
+
+import javax.inject.Inject;
+import javax.inject.Singleton;
+import java.util.List;
+
+@Log
+@Singleton
+public class EndTurnService {
+
+  private final GameRoom gameRoom;
+  private final GameRoomSyncer gameRoomSyncer;
+  private String currentToken;
+
+  private final Json json = new Json();
+
+  @Inject
+  EndTurnService(
+      @NonNull World world,
+      @NonNull ComponentFactory componentFactory,
+      @NonNull GameRoom gameRoom,
+      @NonNull GameRoomSyncer gameRoomSyncer
+  ) {
+    this.gameRoom = gameRoom;
+    this.gameRoomSyncer = gameRoomSyncer;
+  }
+
+  public void init() {
+    this.currentToken = gameRoom.getClient(0).getPlayerToken();
+  }
+
+  public void nextTurn(@NonNull Client client) {
+    if (client.getPlayerToken().equals(currentToken)) {
+      var clients = gameRoom.getClients().stream().toList();
+      for (int i = 0; i < clients.size(); i++) {
+        if (clients.get(i).getPlayerToken().equals(client.getPlayerToken())) {
+          giveControlToPlayer(i + 1, clients);
+        }
+      }
+    } else {
+      log.info("Player " + client.getPlayerUsername() + " tried to end turn");
```

Comment:
jakby blokować input na poziomie gracza to nie powinno się to zdarzyć

Reply:
ale lepiej to dać, bo na pewno będą jakieś problemy w dalszej rozgrywce wieloosobowej